### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by SternAndrew**
Allow for cleanup of the ServerConnection object and to be more consistent with the StreamSocket and session pointer objects.  Also added try...catch around cleanup since Poco can throw exceptions but destructors are noexcept by default in c++ x11 so not catching exceptions here can bring down the application.

### CODE REVIEW

File: include/fix8/sessionwrapper.hpp [hunk 1]
```diff
@@ -646,10 +646,39 @@ class SessionInstance : public SessionInstanceBase
...
+		try
+		{
+			if (_psc != nullptr)
+			{
+				_psc->stop();
+				delete _psc;
+				_psc = nullptr;
+			}
+		}
+		catch (...)
```
Comment (dakka) [inline include/fix8/sessionwrapper.hpp]: An exception catch-all is not suitable here. You need to at least specify the base class of the exception you are expecting to catch.

File: include/fix8/sessionwrapper.hpp [hunk 2]
```diff
@@ -646,10 +646,39 @@ class SessionInstance : public SessionInstanceBase
...
+		{
+			std::cerr << "~SessionInstance - psc exception during cleanup" << std::endl;;
```
Comment (dakka) [inline include/fix8/sessionwrapper.hpp]: Printing a message to `cerr` is not suitable here. You need to send an error message to one of the loggers, such as

    this->_session->log(e.what(), Logger::Error);

**Comment by dakka** [comment]
`./schema/TSX_EXCHANGE_FIX43.xml` contains customised FIX field types that are non-standard. If you wish to contribute these then place them in `./contrib` with a note in the header about why these changes were made, what they are for.

**Comment by dakka** [comment]
@SternAndrew are you going to make the suggested changes so this pull request can be merged?

**Comment by SternAndrew** [comment]
I'll be making the change.  Needed to work on higher priority items outside of fix8.

________________________________
From: David Dight
Sent: Friday, June 19, 2015 4:06 AM
To: fix8/fix8
CC: "Stern, Andrew"
Subject: Re: [fix8] Dev (#149)



@SternAndrew<https://github.com/SternAndrew> are you going to make the suggested changes so this pull request can be merged?

—
Reply to this email directly or view it on GitHub<https://github.com/fix8/fix8/pull/149#issuecomment-113423923>.
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-

This message is for the named person's use only. This communication is for informational purposes only and has been obtained from sources believed to be reliable, but it is not necessarily complete and its accuracy cannot be guaranteed. It is not intended as an offer or solicitation for the purchase or sale of any financial instrument or as an official confirmation of any transaction. Moreover, this material should not be construed to contain any recommendation regarding, or opinion concerning, any security. It may contain confidential, proprietary or legally privileged information. No confidentiality or privilege is waived or lost by any mistransmission. If you receive this message in error, please immediately delete it and all copies of it from your system, destroy any hard copies of it and notify the sender. You must not, directly or indirectly, use, disclose, distribute, print, or copy any part of this message if you are not the intended recipient. Any views expressed in this message are those of the individual sender, except where the message states otherwise and the sender is authorized to state them to be the views of any such entity.

Securities products and services provided to Canadian investors are offered by ITG Canada Corp. (member CIPF and IIROC - Investment Industry Regulatory Organization of Canada), an affiliate of Investment Technology Group, Inc.

Investment research products and services are produced and offered by ITG Investment Research, Inc. and not ITG Inc. (a FINRA member firm and SIPC member).

For CANADIANS Only: In accordance with the Canadian AntiSpam Legislation, ITG Canada has implied consent from you as a member of the investment community with whom we have already established a relationship through business discussions or dealings, or your email address was made available to ITG. However, if you are Canadian and wish to stop receiving ANY emails from ITG, please click on this button Unsubscribe <mailto:can-casl@itg.com?subject=Unsubscribe>. Please note that this does not apply if you are an existing client, and IIROC & OSC rules & regulations require us to continue to send you critical email communications

ITG Inc. and/or its affiliates reserves the right to monitor and archive all electronic communications through its network.

ITG Inc. Member FINRA, SIPC

-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-

**Pull request merged**